src/gui/guiFormSpecMenu.cpp

```diff
@@ -3307,54 +3312,74 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 		case BET_DOWN:
 			// Some mouse button has been pressed
 
-			//infostream<<"Mouse button "<<button<<" pressed at p=("
+			//infostream<<"Mouse button "<<m_button<<" pressed at p=("
 			//	<<p.X<<","<<p.Y<<")"<<std::endl;
 
 			m_selected_dragging = false;
 
 			if (s.isValid() && s.listname == "craftpreview") {
 				// Craft preview has been clicked: craft
-				craft_amount = (button == BET_MIDDLE ? 10 : 1);
+				craft_amount = (m_button == BET_MIDDLE ? 10 : 1);
 			} else if (!m_selected_item) {
-				if (s_count && button != BET_WHEEL_UP) {
+				if (s_count && m_button != BET_WHEEL_UP) {
 					// Non-empty stack has been clicked: select or shift-move it
 					m_selected_item = new ItemSpec(s);
 
 					u32 count;
-					if (button == BET_RIGHT)
+					if (m_button == BET_RIGHT)
 						count = (s_count + 1) / 2;
-					else if (button == BET_MIDDLE)
+					else if (m_button == BET_MIDDLE)
 						count = MYMIN(s_count, 10);
-					else if (button == BET_WHEEL_DOWN)
+					else if (m_button == BET_WHEEL_DOWN)
 						count = 1;
 					else  // left
 						count = s_count;
 
 					if (!event.MouseInput.Shift) {
 						// no shift: select item
+
+						// m_selected_item can be deleted by the black magic when it's empty
+						// As such we create our own copy of it for left drag handling
+						m_left_drag_item = new ItemSpec(s);
```

Comment:
This line is executed when taking an item stack from the inventory but is not executed when taking the item stack from the crafting output.

Reply:
You're right, thanks for noticing. Though it's unclear why the author used `m_left_drag_item` and not `m_selected_item` only.